File: src/etch/bindings/java/msg/test/TestTypeValidator.java
package etch.bindings.java.msg.test;

import static org.junit.Assert.*;

import org.junit.Test;

import etch.bindings.java.msg.TypeValidator;
import etch.bindings.java.msg.Validator;

/** */
public class TestTypeValidator
{
	/** */
	@Test( expected = IllegalArgumentException.class )
	public void testValidate()
	{
		new MyTypeValidator( int.class, int.class, -1, "bzzt" );
	}

	/** */
	@Test
	public void testTypeValidator1()
	{
		MyTypeValidator v = new MyTypeValidator( int.class, null, 0, "abc" );
		assertSame( int.class, v.getExpectedClass() );
		assertEquals( 0, v.getNDims() );
		assertEquals( "abc", v.toString() );
	}

	/** */
	@Test
	public void testTypeValidator2()
	{
		MyTypeValidator v = new MyTypeValidator( boolean.class, null, 0, "def" );
		assertSame( boolean.class, v.getExpectedClass() );
		assertEquals( 0, v.getNDims() );
		assertEquals( "def", v.toString() );
	}

	/** */
	@Test
	public void testTypeValidator3()
	{
		MyTypeValidator v = new MyTypeValidator( null, int.class, 1, "ghi" );
		assertSame( int[].class, v.getExpectedClass() );
		assertEquals( 1, v.getNDims() );
		assertEquals( "ghi", v.toString() );
	}

	/** */
	@Test
	public void testTypeValidator4()
	{
		MyTypeValidator v = new MyTypeValidator( null, boolean.class, 1, "jkl" );
		assertSame( boolean[].class, v.getExpectedClass() );
		assertEquals( 1, v.getNDims() );
		assertEquals( "jkl", v.toString() );
	}

	/** */
	@Test
	public void testToString()
	{
		MyTypeValidator v = new MyTypeValidator( int.class, int.class, 0, "abc" );
		assertEquals( "abc", v.toString() );
		
		v = new MyTypeValidator( int.class, int.class, 0, "def" );
		assertEquals( "def", v.toString() );
		
		v = new MyTypeValidator( int.class, int.class, 0, "ghi" );
		assertEquals( "ghi", v.toString() );
	}

	/** @throws Exception */
	@Test
	public void testSilly() throws Exception
	{
		new MyTypeValidator( int.class, int.class, 0, "ghi" ).checkValue( 0 );
		new MyTypeValidator( int.class, int.class, 0, "ghi" ).elementValidator();
	}

	/** */
	public static class MyTypeValidator extends TypeValidator
	{
		/**
		 * @param scalarClass
		 * @param arrayClass
		 * @param nDims
		 * @param descr
		 */
		public MyTypeValidator( Class<?> scalarClass, Class<?> arrayClass,
			int nDims, String descr )
		{
			super( scalarClass, arrayClass, nDims, descr );
			// TODO Auto-generated constructor stub
		}

		@Override
		public Byte checkValue( Object value )
		{
			// TODO Auto-generated method stub
			return null;
		}

		@Override
		public Validator elementValidator()
		{
			// TODO Auto-generated method stub
			return null;
		}
	}
}

File: src/etch/bindings/csharp/msg/test/TestTypeValidator.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;

namespace Etch.Msg.Test
{
    [TestFixture]
    public class TestTypeValidator
    {

        [Test]
        [ExpectedException( typeof( ArgumentOutOfRangeException ) )]
        public void TestValidate()
        {
            new MyTypeValidator( typeof( int ), typeof( int ), -1, "bzzt" );
        }

        [Test]
        public void TestTypeValidator1()
        {
            MyTypeValidator v = new MyTypeValidator( typeof( int ), null, 0, "abc" );
            Assert.AreEqual( typeof( int ), v.GetExpectedClass() );
            Assert.AreEqual( 0, v.GetNDims() );
            Assert.AreEqual( "abc", v.ToString() );
        }

        [Test]
        public void TestTypeValidator2()
        {
            MyTypeValidator v = new MyTypeValidator( typeof(Boolean), null, 0, "def" );
            Assert.AreEqual( typeof( Boolean ), v.GetExpectedClass() );
		    Assert.AreEqual( 0, v.GetNDims() );
		    Assert.AreEqual( "def", v.ToString() );
        }

        [Test]
	    public void TestTypeValidator3()
	    {
		    MyTypeValidator v = new MyTypeValidator( null, typeof(int), 1, "ghi" );
		    Assert.AreEqual( typeof(int[]), v.GetExpectedClass() );
		    Assert.AreEqual( 1, v.GetNDims() );
		    Assert.AreEqual( "ghi", v.ToString() );
	    }

        [Test]
	    public void TestTypeValidator4()
	    {
		    MyTypeValidator v = new MyTypeValidator( null, typeof(Boolean), 1, "jkl" );
		    Assert.AreEqual( typeof(Boolean[]), v.GetExpectedClass() );
		    Assert.AreEqual( 1, v.GetNDims() );
		    Assert.AreEqual( "jkl", v.ToString() );
	    }

        [Test]
	    public void TestToString()
	    {
		    MyTypeValidator v = new MyTypeValidator( typeof(int), typeof(int), 0, "abc" );
		    Assert.AreEqual( "abc", v.ToString() );
    		
		    v = new MyTypeValidator( typeof(int), typeof(int), 0, "def" );
		    Assert.AreEqual( "def", v.ToString() );
    		
		    v = new MyTypeValidator( typeof(int), typeof(int), 0, "ghi" );
		    Assert.AreEqual( "ghi", v.ToString() );
	    }

        [Test]
	    public void TestSilly()
	    {
		    new MyTypeValidator( typeof(int), typeof(int), 0, "ghi" ).CheckValue( 0 );
		    new MyTypeValidator( typeof(int), typeof(int), 0, "ghi" ).ElementValidator();
	    }

        public class MyTypeValidator : TypeValidator
        {
            public MyTypeValidator( Type scalarClass, Type arrayClass, int nDims,
                String descr ) : base ( scalarClass, arrayClass, nDims, descr)
            { }

            public override sbyte? CheckValue( Object value )
            {
                // ignore
                return null;
            }

            public override Validator ElementValidator()
            {
                // ignore
                return null;
            }
        }
    }
}
